### ISSUE TITLE
Question about WASM

### ISSUE THREAD

**Issue by Slixe** [fzyzcjy/flutter_rust_bridge#2087]
Hey there,

We're using daily your crate so thanks a lot for it.

We would like to expand the supported devices to web also, unfortunately I struggle to find enough information for that.

Using `flutter_rust_bridge_serve --crate .` compilation is ok but once it has finished, I get the following:
```
error: cannot import from modules (`env`) with `--no-modules`
Error: Running the wasm-bindgen CLI
Caused by: Running the wasm-bindgen CLI
Caused by: failed to execute `wasm-bindgen`: exited with exit status: 1
```

Is WASM actually supported ? I don't use the env myself, maybe a lib does, how can I check that and, if its for web target, shouldn't it be `--target web` instead of `--no-modules` ?

As an example, `wasm-pack build --target web --no-default-features` is working fine but I have no idea how to glue it and run it in my browser directly with flutter.

Thanks in advance

**Comment by fzyzcjy** [comment]
You are welcome and happy to see it is useful for you!

To run in web, could you please try https://cjycode.com/flutter_rust_bridge/quickstart#3-run-it, i.e. copying from there -

```
flutter_rust_bridge_codegen build-web
# ... or any other standard Flutter ways
flutter run --web-header=Cross-Origin-Opener-Policy=same-origin --web-header=Cross-Origin-Embedder-Policy=require-corp
```

**Comment by Slixe** [comment]
Here is the log using your command:

```
FRB_DART_RUN_COMMAND_STDERR=1 flutter_rust_bridge_codegen build-web

> which wasm-pack (pwd: null, env: null)
/home/slixe/.cargo/bin/wasm-pack
> cargo read-manifest (pwd: rust, env: null)
> wasm-pack build -t no-modules -d /home/slixe/Projects/XELIS/xelis_genesix_wallet/web/pkg --no-typescript --out-name rust_lib --dev rust -- -Z build-std=std,panic_abort (pwd: null, env: {RUSTUP_TOOLCHAIN: nightly, RUSTFLAGS: -C target-feature=+atomics,+bulk-memory,+mutable-globals, CARGO_TERM_COLOR: always})
^[[INFO]: 🎯  Checking for the Wasm target...
[INFO]: 🌀  Compiling to Wasm...
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.13s
error: cannot import from modules (`env`) with `--no-modules`
Error: Running the wasm-bindgen CLI
Caused by: Running the wasm-bindgen CLI
Caused by: failed to execute `wasm-bindgen`: exited with exit status: 1
  full command: "/home/slixe/.cargo/bin/wasm-bindgen" "/home/slixe/Projects/XELIS/xelis_genesix_wallet/rust/target/wasm32-unknown-unknown/debug/rust_lib.wasm" "--out-dir" "/home/slixe/Projects/XELIS/xelis_genesix_wallet/web/pkg" "--no-typescript" "--target" "no-modules" "--out-name" "rust_lib" "--debug"
Unhandled exception:
ProcessException: Bad exit code (1). stderr=[INFO]: 🎯  Checking for the Wasm target...
[INFO]: 🌀  Compiling to Wasm...
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.13s
error: cannot import from modules (`env`) with `--no-modules`
Error: Running the wasm-bindgen CLI
Caused by: Running the wasm-bindgen CLI
Caused by: failed to execute `wasm-bindgen`: exited with exit status: 1
  full command: "/home/slixe/.cargo/bin/wasm-bindgen" "/home/slixe/Projects/XELIS/xelis_genesix_wallet/rust/target/wasm32-unknown-unknown/debug/rust_lib.wasm" "--out-dir" "/home/slixe/Projects/XELIS/xelis_genesix_wallet/web/pkg" "--no-typescript" "--target" "no-modules" "--out-name" "rust_lib" "--debug"

  Command: wasm-pack build -t no-modules -d /home/slixe/Projects/XELIS/xelis_genesix_wallet/web/pkg --no-typescript --out-name rust_lib --dev rust -- -Z build-std=std,panic_abort
#0      runCommand (package:flutter_rust_bridge/src/cli/run_command.dart:67:5)
<asynchronous suspension>
#1      _executeWasmPack (package:flutter_rust_bridge/src/cli/build_web/executor.dart:143:3)
<asynchronous suspension>
#2      executeBuildWeb (package:flutter_rust_bridge/src/cli/build_web/executor.dart:59:3)
<asynchronous suspension>
#3      BuildWebCommand.run (package:flutter_rust_bridge/src/cli/build_web/entrypoint.dart:18:5)
<asynchronous suspension>
#4      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#5      main (file:///home/slixe/.pub-cache/hosted/pub.dev/flutter_rust_bridge-2.0.0-dev.32/bin/flutter_rust_bridge.dart:8:3)
<asynchronous suspension>
Error: Fail to execute command, please see logs above for details.
```

_In case, the project is hosted here: https://github.com/xelis-project/xelis-genesix-wallet in branch `wasm-test`_
It is not a small project unfortunately, I will try to reproduce a minimal version if I can..

**Comment by fzyzcjy** [comment]
Ok I see. Seems someone has reported before: https://github.com/fzyzcjy/flutter_rust_bridge/issues/1915. Could you please check whether it has similar scenarios as yours?

**Comment by fzyzcjy** [comment]
To debug which dependency causes this, I guess we can do something like: Firstly temporarily remove all real source code (to avoid them using dependencies and causing compile error), and them bisect the dependencies to know which exact one.

**Comment by Slixe** [comment]
I've found the cause of this env error, I'm using `SledDB` in `xelis_wallet` dependency, which cause this error. I did a mock DB just to test with the `#[wasm_bindgen]` on my `Wallet` struct and its working.

Now, back to flutter rust bridge with this change, looks like I can build with no issues using
`flutter_rust_bridge_codegen build-web`.

In summary, WASM is well compatible and is working correctly! The problem was on my side and not on your, sorry for this issue.

Thanks again for your amazing lib!
